client/dataset.go

```diff
@@ -33,8 +37,16 @@ var _ Datasets = (*datasets)(nil)
 //
 // API docs: https://docs.honeycomb.io/api/dataset
 type Dataset struct {
-	Name string `json:"name"`
-	Slug string `json:"slug,omitempty"`
+	Name            string  `json:"name"`
+	Description     *string `json:"description,omitempty"`
+	Slug            string  `json:"slug,omitempty"`
+	ExpandJSONDepth *int    `json:"expand_json_depth,omitempty"`
+}
+
+type DatasetCreateArgs struct {
```

Comment:
I would rather see `Dataset` follow the same update pattern as all of the other resources in the provider's client instead of this approach with the `CreateArgs` struct. It may not be ideal the way it is done, but there is something to be said for consistency in a codebase.